src/main/java/com/shapesecurity/shift/scope/ScopeAnalyzer.java

```diff
@@ -364,81 +363,26 @@ private State() {
      */
     @SuppressWarnings({"AccessingNonPublicFieldOfAnotherObject", "ObjectEquality"})
     private State(@NotNull State a, @NotNull State b) {
-      if (a == b) {
```

Comment:
That is moved to `Monoid.append` to avoid creating a new State in the first place.
at https://github.com/shapesecurity/shift-java/pull/45/files#diff-8dbcb83e57e5eef5979d0f460fad5ac8R609